### DISCUSSION TITLE
Is there a way to calculate API parameters in Budibase REST API data source connections?

### DISCUSSION THREAD

**Discussion by starsang1995** [Budibase/budibase#5891] [Help] [answered]
Hi we would like to integrate with a third party platform and since they are using a calculated query parameter in the URL as authentication method, so I took a look into the REST API in data source but I didn't find a way to execute code and add the result into the parameter before making the request.
Since I'm new to Budibase, I'm not sure if there is a such function but I didn't see, or maybe there is not. So kindly ask if someone know this. Many thanks!

**Comment by melohagan** [comment, marked as answer]
Hey @apsonandtesla 

It is possible to pass a binding into a REST Query: https://docs.budibase.com/docs/rest-bindings

Instead of using handlebars, you can use JavaScript bindings which will give you more flexibility in code execution.

**Reply by starsang1995** [reply]
Thanks for the reply! Hmm sure it should do the trick but it doesn't sound very intuitive but more like a workaround. Thanks anyway!
